topydo/lib/ChangeSet.py

```diff
@@ -0,0 +1,94 @@
+# Topydo - A todo.txt client written in Python.
+# Copyright (C) 2014 - 2015 Bram Schoenmakers <me@bramschoenmakers.nl>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" This module serves for managing todo and archive changesets. """
+
+from topydo.lib.Config import config
+from hashlib import sha1
+import shelve
+from copy import deepcopy
+
+class ChangeSet(object):
+    def __init__(self, p_todolist=None, p_archive=None):
+        self.todolist = deepcopy(p_todolist)
+        self.todolist_hash = self._hash_todolist(p_todolist)
+        self.archive = deepcopy(p_archive)
+        self.archive_hash = self._hash_todolist(p_archive)
+        self.shelve_file = shelve.open(config().backup_file())
```

Comment:
The path of the backup file should be relative to the active todo.txt file. Otherwise things get messy with multiple todo.txt files, or when running topydo from different working directories.